rb/spec/integration/selenium/webdriver/manager_spec.rb

```diff
@@ -53,103 +53,221 @@ module WebDriver
+          expect(driver.manage.cookie_named('domain')[:domain]).to eq('.saucelabs.com')
         end
 
-        it 'should add expiration with Time' do
-          driver.navigate.to url_for('xhtmlTest.html')
+        it 'should not allow domain to be set for localhost',
+           exclude: {browser: :chrome,
+                     reason: "https://bugs.chromium.org/p/chromedriver/issues/detail?id=3733"} do
+          expect {
+            driver.manage.add_cookie name: 'domain',
+                                     value: 'localhost',
+                                     domain: 'localhost'
+          }.to raise_error(Error::UnableToSetCookieError)
+        end
 
-          expected = (Date.today + 2).to_datetime
-          driver.manage.add_cookie name: 'expiration',
-                                   value: 'time',
-                                   expires: expected.to_time
+        it 'should not allow setting on a different domain',
+           except: {browser: :chrome,
+                    reason: "https://bugs.chromium.org/p/chromedriver/issues/detail?id=3734"} do
+          expect {
+            driver.manage.add_cookie name: 'domain',
+                                     value: 'different',
+                                     domain: 'selenium.dev'
+          }.to raise_error(Error::InvalidCookieDomainError)
+        end
 
-          actual = driver.manage.cookie_named('expiration')[:expires]
-          expect(actual).to be_kind_of(DateTime)
-          expect(actual).to eq(expected)
+        it 'should not allow setting on a subdomain from parent domain',
+           exclusive: {driver: :none,
+                       reason: "Can only be tested on site with subdomains"},
+           except: {browser: :chrome,
+                    reason: 'https://bugs.chromium.org/p/chromedriver/issues/detail?id=3734'} do
+          driver.get("https://saucelabs.com")
+
+          expect {
+            driver.manage.add_cookie name: 'domain',
+                                     value: 'subdomain',
+                                     domain: 'opensource.saucelabs.com'
+          }.to raise_exception(Error::InvalidCookieDomainError)
         end
 
-        it 'should add expiration with Number' do
-          driver.navigate.to url_for('xhtmlTest.html')
+        it 'should not be visible to javascript when http_only is true' do
+          driver.manage.add_cookie name: 'httponly',
+                                   value: 'true',
+                                   http_only: true
+
+          expect(driver.execute_script("return document.cookie")).to be_empty
+          expect(driver.manage.cookie_named('httponly')[:http_only]).to eq true
+        end
 
-          expected = (Date.today + 2).to_datetime
-          driver.manage.add_cookie name: 'expiration',
-                                   value: 'number',
-                                   expires: expected.to_time.to_f
+        it 'should not add secure cookie when http',
```

Comment:
I don't understand this test. Although maybe I'm missing a setup step. Shouldn't this first navigate to something like `http://insecure.com/` and then attempt to set a secure cookie?

Reply:
it starts up in a non-secure protocol, but sure might as well add navigation first.

Reply:
oh yeah, I state http in the spec title, but then don't navigate to http 👍